api/v1alpha1/lease_helpers.go

```diff
@@ -73,12 +74,107 @@ func ReconcileLeaseTimeFields(beginTime, endTime **metav1.Time, duration **metav
 	return nil
 }
 
+// ParseLabelSelector parses a label selector string and converts it to metav1.LabelSelector.
+// This function supports the != operator by first parsing with labels.Parse() which supports
+// all label selector syntax including !=, then converting to metav1.LabelSelector format.
+func ParseLabelSelector(selectorStr string) (*metav1.LabelSelector, error) {
+	// First, try to parse using labels.Parse() which supports != operator
+	parsedSelector, err := labels.Parse(selectorStr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse label selector: %w", err)
+	}
+
+	// Extract requirements from the parsed selector
+	requirements, selectable := parsedSelector.Requirements()
+	if !selectable {
+		return &metav1.LabelSelector{}, nil
+	}
+
+	// Convert requirements to metav1.LabelSelector format
+	matchLabels := make(map[string]string)
+	var matchExpressions []metav1.LabelSelectorRequirement
+
+	// Track NotEquals requirements by key so we can combine them into NotIn
+	notEqualsByKey := make(map[string][]string)
+
+	for _, req := range requirements {
+		key := req.Key()
+		operator := req.Operator()
+		values := req.ValuesUnsorted()
+
+		switch operator {
+		case selection.Equals:
+			// For exact match with single value, use matchLabels
+			if len(values) == 1 {
+				// Check if we already have an equality requirement for this key with a different value
+				if existingValue, exists := matchLabels[key]; exists && existingValue != values[0] {
+					return nil, fmt.Errorf("invalid selector: label %s cannot have multiple equality requirements with different values (%s and %s)", key, existingValue, values[0])
+				}
+				matchLabels[key] = values[0]
+			} else {
+				// Multiple values should use In operator
```

Comment:
possibly redundant
https://pkg.go.dev/k8s.io/apimachinery/pkg/labels#NewRequirement
> If the operator is Equals, DoubleEquals, or NotEquals, the values set must contain one value.

Reply:
Ahh no, while looking at this code it seems correct. the branch in L108 handles repeated keys, it will fail with error: invalid selector: label %s cannot have multiple equality requirements with different values (%s and %s)"

The 2nd branch handles the case where

a=[X,Y,Z],

not sure how you write it in their parseable format, but it means that a can be X, Y or Z, so they use an "a in (X, Y, Z)" selector.

I think we are fine ! :) please have a look.